apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/status/StatusChecker.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.context.status;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.apache.skywalking.apm.agent.core.conf.Config;
+
+public enum StatusChecker {
+
+    /**
+     * All exception is thought as error status.
+     */
+    OFF(Collections.singletonList(new OffExceptionCheckStrategy()), new ArrayList<>()),
+
+    /**
+     * 1. The throwable is not instance of exception, return error status. <br/> 2. The exception name is exist in
+     * ignoreExceptions, return normal status.<br/>
+     *
+     * @see Config.StatusCheck#IGNORED_EXCEPTIONS
+     */
+    NAMED_MATCH(
+        Collections.singletonList(new FastFailExceptionCheckStrategy()),
+        Arrays.asList(
+            new NamedMatchExceptionCheckStrategy()
+        )
+    ),
+    /**
+     * 1. The throwable is not instance of exception, return error status. <br/> 2. The exception name is exist in
+     * ignoreExceptions, return normal status.<br/> 3. The exception superclass name is exist in ignoreExceptions,
+     * return normal status.<br/>
+     *
+     * @see Config.StatusCheck#IGNORED_EXCEPTIONS
+     */
+    INHERIT_NAMED_MATCH(
+        Collections.singletonList(new FastFailExceptionCheckStrategy()),
+        Arrays.asList(
+            new NamedMatchExceptionCheckStrategy(),
+            new InheriteMatchExceptionCheckStrategy()
```

Comment:
If I read the codes right, InheriteMatchExceptionCheckStrategy should have included the logic of `NamedMatchExceptionCheckStrategy`.